demo: move markdown/prism rendering to runtime

maxokorokov opened the pull request:
We need code highlighting for documentation/overview pages. This PR introduces 
* `marked` and `prism` to application runtime
* `<ngbd-code>` component for highlighted code blocks
* deprecates `prism-loader` for the demos - (uncompressed bundle is 600Kb smaller and perf is actually a bit better when doing highlighting at runtime)

pkozlowski-opensource commented on demo/src/app/components/shared/example-box/example-box.component.html:
```diff
@@ -36,7 +36,7 @@ <h2>
         </ng-template>
         <ng-template ngbTabContent>
-          <pre class="language-typescript"><code class="language-typescript" [innerHTML]="snippets[demo]['code']"></code></pre>
+          <ngbd-code [code]="snippets[demo]['code']" lang="javascript"></ngbd-code>
```
Why the change from `typescript ` => `javascript `?

pkozlowski-opensource commented on demo/src/app/shared/markdown/code.component.ts:
```diff
@@ -0,0 +1,23 @@
+
+  ngOnInit() {
+    const code = `\`\`\`${this.lang}\n${this.code.trim()}\n\`\`\``;
+    this.el.nativeElement.innerHTML = this.service.toHTML(code);
```
Not sure why do we pass code through markdown service?

maxokorokov replied:
Oh, wow, got a little excited there playing with markdown. Will remove markdown completely from this PR and leave only code highlighting component

maxokorokov commented:
Thanks for the review, updated:
- removed markdown completely from this PR
- added `typescript` and `bash` language highlights
- updated `code.component` and `code-highlight.service`
- fixed issue when prism was trying to re-highlighting everything on DOM content load
